backend/src/index.ts

```diff
@@ -80,17 +81,25 @@ async function main() {
       // We know that block number will not exceed 53-bit size integer
       let lastProcessedBlock = Number(totals.count.toBigInt()) - 1;
 
+      const relayParams = {
+        logger,
+        target,
+        batchBytesLimit: BATCH_BYTES_LIMIT,
+      };
+
+      let relay;
+
+      // Relay accepts different parameters depending on the current mode: process blocks from archive or over the network
       if (chainConfig.downloadedArchivePath) {
+        const archive = new ChainArchive({ path: chainConfig.downloadedArchivePath, logger });
+        relay = new Relay({ ...relayParams, archive, batchCountLimit: ARCHIVE_BATCH_COUNT_LIMIT });
+
         try {
-          lastProcessedBlock = await relayFromDownloadedArchive(
+          lastProcessedBlock = await relay.fromDownloadedArchive(
```

Comment:
What about having `archive` as an argument in this method and simplify `relay` constructor?